### PR TITLE
Feature/cvsb 19017

### PR DESCRIPTION
**Pull request by JonLazariniDD**
configure repo for development

### CODE REVIEW

**Comment by JonLazariniDD** [comment]
I was thinking that if we wish to replicate this scaffold project we could leave the history until 1.0.0 which should be our starting point from releasing changes now since the configuration is done as this would help cherry picking (or not) changes if we didn't want to start with the same baseline or modifying this further? Thoughts?

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# https://docs.npmjs.com/cli/v6/commands/npm-audit
+audit-level = high
+audit = true
+init-version = "1.0.0"
+# save exact dependencies in case some do not rely on semver toa void dependencies hell
+save-exact = true
+sign-git-commit = false
+# handled by semantic release - format is x.x.x compying with semver standards
+# tag-version-prefix
+versions = true
```
Comment (m-ujaffer) [inline .npmrc:13]: linebreak

File: .nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+12.5.0
```
Comment (m-ujaffer) [inline .nvmrc:1]: linebreak

File: .prettierignore [hunk 3]
```diff
@@ -0,0 +1,5 @@
+node_modules
+.serverless
+.webpack
+.artifact
+.build
```
Comment (m-ujaffer) [inline .prettierignore:5]: linebreak

File: .secignore [hunk 4]
```diff
@@ -0,0 +1 @@
+.npmrc
```
Comment (m-ujaffer) [inline .secignore:1]: linebreak

File: .editorconfig [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+# Matches multiple files with brace expansion notation
+# Set default charset
+[*.{js,ts, *.json, *.yml}]
+charset = utf-8
+indent_style = space
+indent_size = 2
+
+# Tab indentation (no size specified)
+[Makefile]
+indent_style = tab
```
Comment (m-ujaffer) [inline .editorconfig:19]: line break

File: .eslintignore [hunk 6]
```diff
@@ -0,0 +1,7 @@
+node_modules
+.serverless
+.vscode
+*.config.js
+.webpack
+**/*.js
+.build
```
Comment (m-ujaffer) [inline .eslintignore:7]: linebreak

File: package.json [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    "lint:analyse": "eslint . --ext js,ts --fix",
+    "lint:report": "npm run lint:analyse -- -f json -o reports/eslint/eslint-report.json",
+    "lint": "npm-run-all lint:*",
+    "prettier": "prettier --write './**/*.{js,ts}'",
+    "security-checks": "git secrets --scan && git log -p | scanrepo",
+    "clean": "rimraf ./.build ./.artifact ./.serverless ./.webpack ./*.zip",
+    "compile": "tsc",
+    "build:dev": "cross-env API_VERSION=${npm_package_version} NODE_ENV=local serverless webpack",
+    "build:production": "webpack -c ./scripts/webpack.config.build.js --env=production",
+    "build": "npm-run-all clean build:dev",
```
Comment (m-ujaffer) [inline package.json:39]: nice

File: README.md [hunk 8]
```diff
@@ -1 +1,208 @@
...
+</br>
+
+### Scripts
+
+The following scripts are available, for further information please refer to the project `package.json` file:
+
+- <b>start</b>: `npm start` - _launch serverless offline service_
+- <b>dev</b>: `npm run dev` - _run in parallel the service and unit tests in_ `--watch` _mode with live reload_.
+- <b>test</b>: `npm t` - _execute the unit test suite_
+- <b>build</b>: `npm run build` - _bundle the project for production_
```
Comment (m-ujaffer) [inline README.md:67]: minor tweak needed as production and dev build are two separate scripts.

**Review by m-ujaffer — CHANGES_REQUESTED** [review]
A very good repository and can be used to build new services. I would recommend to strip out some keywords specific to trailer registration and setup as a scaffolding which can be cloned to make new micro-services. There are a few minor which need to be looked at.

**Comment by m-ujaffer** [comment]
closing in favour of https://github.com/dvsa/cvs-svc-template

**Pull request closed without merging**